Создание классов

AlexandraLis295 opened the pull request:
1. Создали классы, соответствующие сущностям базы данных прошлого семестра.
2. Создали в классах несколько объектов, соответствующих палям таблиц базы данных из прошлого семестра.
3. Создали конструкторы к соответствующим классам.
4. В каждом классе, соответствующем полям, переопределили метод  "ToString".
5. Создали класс, где предусмотрели главный метод и вывод тестовых значений на консоль.
6. Первая попытка создать зачатки документации.

AlexandraLis295 committed:
1. Создали классы, соответствующие сущностям базы данных прошлого семестра.

2. Создали в классах несколько объектов, соответствующих палям таблиц базы данных из прошлого семестра.
3. Создали конструкторы к соответствующим классам.
4. В каждом классе, соответствующем полям, переопределили метод  "ToString".
5. Создали класс, где предусмотрели главный метод и вывод тестовых значений на консоль.
6. Первая попытка создать зачатки документации.

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Operators.cs:
```diff
@@ -0,0 +1,30 @@
+        /// </summary>
+        /// <param name="id">ID</param>
+        /// <param name="NameOfOperator">Название оператора</param>
+        public Operators(int id, string NameOfOperator)
```
локальная переменная - с маленькой буквы

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Operators.cs:
```diff
@@ -0,0 +1,30 @@
+        public Operators(int id, string NameOfOperator)
+        {
+            ID = id;
+            OperatorName = NameOfOperator;
```
все же используйте явно указатель `this`
Сделайте проверку на то, что имя не пустое и не `null` (спишите у меня)

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Operators.cs:
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
уберите неиспользуемые библиотеки

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Operators.cs:
```diff
@@ -0,0 +1,30 @@
+﻿using System;
```
лучше переместить на 6 строку под `namespace`

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Program.cs:
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
```
уберите неиспользуемые `using`, используемые переместите после `namespace`

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Services.cs:
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
```
также

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Subscribers.cs:
```diff
@@ -0,0 +1,53 @@
+        public Subscribers(int id, string lastName, string firstName, string patronymic, int yearOfBirth, int monthOfBirth, int dayOfBirth)
+        {
+            ID = id;
+            if (lastName == null)
+                throw new ArgumentNullException(nameof(lastName), "LastName cannot be null");
+            LastName = lastName;
+            if (firstName == null)
```
могут прийти и пустые, их тоже нужно проверять

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Subscribers.cs:
```diff
@@ -0,0 +1,53 @@
+                throw new ArgumentNullException(nameof(firstName), "FirstName cannot be null");
+            FirstName = firstName;
+            Patronymic = patronymic;
+            YearOfBirth = yearOfBirth;
+            MonthOfBirth = monthOfBirth;
+            DayOfBirth = dayOfBirth;
+        }
```
откажитесь от такой точности или создавайте класс дата((( (в `C#`, кажется что есть...

MarinaPaley commented on CommunicationServicesORM/CommunicationServicesORM/Subscribers.cs:
```diff
@@ -0,0 +1,53 @@
+        /// <returns>Возвращает строковое представление значения, представляемого объектом. Учтена возможность отсутствия у абонента отчества</returns>
+        public override string ToString()
+        {
+            if (Patronymic == null)
+                return String.Format("ID: {0}, Фамилия: {1}, Имя: {2}, Дата рождения: {3}.{4}.{5}\n", ID, LastName, FirstName, YearOfBirth, MonthOfBirth, DayOfBirth);
+            return String.Format("ID: {0}, Фамилия: {1}, Имя: {2}, Отчество: {3}, Дата рождения: {4}.{5}.{6}\n", ID, LastName, FirstName, Patronymic, YearOfBirth, MonthOfBirth, DayOfBirth);
+        }
```
спешите у меня, там был рефакторинг. Самое последнее в репозитории LibraryMagisters

MarinaPaley reviewed (changes requested):
Радует, что начали!!!

AlexandraLis295 committed:
1. Удален класс Operators;

2. Доработаны классы Subscriber и Tariff
3. Добавлены маппинги
4. Добавлена фабрика сессий
5. Реализован CRUD
6. Итого: программа запускается, но падает на фабрике сессий (+ осталось приделать тесты)

AlexandraLis295 committed:
@MarinaPaley

- Программа больше не падает, ура!
- Добавлен тест для тарифов
- Тест проходится без ошибок

AlexandraLis295 committed:
final
